docs/typescript.md

```diff
@@ -47,48 +84,42 @@ const App = () => <Link href="#">Click me</Link>
 
 ### Passing Props
 
-You can type the props of your styled components.\
-Unfortunately, you will need to pass a second parameter with the tag name because TypeScript is unable to infer the tagname.
+You can type the props of your styled components.
 
 ```jsx
 import styled from 'react-emotion'
 
 type ImageProps = {
   src: string,
+  width: number;
 }
 
-const Image = styled<ImageProps, 'div'>('div')`
-  background: url(${props => props.src}) center center;
+const Image0 = styled('div')`
+  width: ${(props: ImageProps) => props.width};
+  background: url(${(props: ImageProps) => props.src}) center center;
   background-size: contain;
 `
-```
 
-### Object Styles
+// Or with object style
 
-```jsx
-import styled from 'react-emotion'
-
-type ImageProps = {
-  src: string,
-}
+const Image1 = styled('div')({
+  backgroundSize: 'contain',
+}, (props: ImageProps) => ({
+  width: props.width;
+  background: `url(${props.src}) center center`,
+}));
 
-const Image = styled<ImageProps, 'div'>('div')({
-  backgroundSize: contain;
-}, ({ src }) => ({
-  background: `url(${src}) center center`,
-}))
+// Or with generic type
 
-// Or shorthand
-
-const Image = styled.div<ImageProps>({
-  backgroundSize: contain;
-}, ({ src }) => ({
-  background: `url(${src}) center center`,
-}))
+const Image1 = styled('div')<ImageProps>({
```

Comment:
This doesn't work, since TS expects a second argument, `styled('div')<ImageProps, 'div'>. TypeScript 2.9 has a fix for this, but 2.9 has not yet been released.